### DISCUSSION TITLE
_AssertionError ('package:firebase_storage/src/reference.dart': Failed assertion: line 127 pos 12: 'file.absolute.existsSync()': is not true.)

### DISCUSSION THREAD

**Discussion by HasancanCakicioglu** [firebase/flutterfire#8076] [Q&A]
this is error = _AssertionError ('package:firebase_storage/src/reference.dart': Failed assertion: line 127 pos 12: 'file.absolute.existsSync()': is not true.)

  Directory appDocDir = await getApplicationDocumentsDirectory();
  String appDocPath = appDocDir.path;
  File file = File(appDocPath);

await FirebaseStorage.instance.ref('uploads/File_file-to-upload').putFile(file!);

i can't upload files to firestorage
the error is in the putFile(file!) section

**Comment by anonymousdomain** [comment]
Future<String> uploadImage(imagefile) async {
    final tempDir = await getTemporaryDirectory();
    final path = '${tempDir.path}/imagefile.jpg';
    final file = File(path);
    UploadTask uploadTask =
        FirebaseStorage.instance.ref().child('post/$postId.jpg').putFile(file);

    String downloadUrl = await (await uploadTask).ref.getDownloadURL();
    return downloadUrl;
  }
I am getting the same error  too

**Comment by AyUsH18102001** [comment]
Any of you got the solution

**Comment by lobnaDesoki** [comment]
did any one get the solution for this error? Because I'm facing the same problem

**Comment by ednurisk** [comment]
is any one solved this error

**Comment by khanzafa** [comment]
just make sure the file parameter exists, as it cannot be empty. you can add a check like the following

```
Future uploadImage(String userId, String petId, String filePath) async {
    try {
      final ref = storage.ref().child('users/$userId/pets/$petId.jpg');
      final file = File(filePath);
      if (!file.existsSync()) {
        throw 'File not found';
      }
      final result = await ref.putFile(File(filePath));
      final url = await result.ref.getDownloadURL();

      return url;
    } catch (e) {
      print('Error uploading image: $e');
    }
  }
```

**Comment by prizzaflutter** [comment]
Remove the app and restart it.
- If the issue persists:
Change the Firebase Storage rule from true to false.
Change the rule back from false to true.
Restart the app. i Hope so??
